### PR TITLE
[Enhance] Add doc for S3DIS semantic segmentation data

### PR DESCRIPTION
**Pull request by Wuziyi616**
Follow #743

### CODE REVIEW

File: docs/datasets/s3dis_sem_seg.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
+# S3DIS for 3D Semantic Segmentation
+
+## Dataset preparation
+
+For the overall process, please refer to the [README](https://github.com/open-mmlab/mmdetection3d/blob/master/data/s3dis/README.md/) page for S3DIS.
+
+### Export S3DIS data
+
+By exporting S3DIS data, we load the raw point cloud data and generate the relevant annotations including semantic label and instance label.
```
Comment (Tai-Wang) [inline docs/datasets/s3dis_sem_seg.md:9]: semantic labels and instance labels.

File: docs/datasets/s3dis_sem_seg.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    xyz_min = np.amin(data_label, axis=0)[0:3]
+    data_label[:, 0:3] -= xyz_min
+
+    np.save(f'{out_filename}_point.npy', data_label[:, :6].astype(np.float32))
+    np.save(f'{out_filename}_sem_label.npy', data_label[:, 6].astype(np.int))
+    np.save(f'{out_filename}_ins_label.npy', data_label[:, 7].astype(np.int))
+
+```
+
+where we load and concatenate all the point cloud instances under `Annotations` to form raw point cloud and generate semantic as well as instance labels. After exporting each room, the point cloud data, semantic labels and instance labels should be saved in `.npy` files.
```
Comment (Tai-Wang) [inline docs/datasets/s3dis_sem_seg.md:94]: and generate semantic/instance labels.
(The overall sentence is a little long and hard to read.)

File: docs/datasets/s3dis_sem_seg.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+]
+```
+
+- `PointSegClassMapping`: Only the valid category ids will be mapped to class label ids like [0, 13) during training. Other class ids will be converted to `ignore_index` which equals to `13`.
+- `IndoorPatchPointSample`: Crop a patch containing a fixed number of points from input point cloud. `block_size` indicates the size of the cropped block, typically `1.0` for S3DIS.
+- `NormalizePointsColor`: Normalize the RGB color values of input point cloud by dividing `255`.
+- Data augmentation:
+    - `GlobalRotScaleTrans`: randomly rotate and scale input point cloud.
+    - `RandomJitterPoints`: randomly jitter point cloud by adding different noise vector to each point.
+    - `RandomDropPointsColor`: setting the colors of point cloud to all zeros by a probability.
```
Comment (Tai-Wang) [inline docs/datasets/s3dis_sem_seg.md:259]: setting -> set

File: docs/datasets/s3dis_sem_seg.md [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+        ann_files=data_root + f's3dis_infos_Area_{test_area}.pkl',
+        pipeline=test_pipeline,
+        classes=class_names,
+        test_mode=True,
+        ignore_index=len(class_names),
+        scene_idxs=data_root +
+        f'seg_info/Area_{test_area}_resampled_scene_idxs.npy'))
+```
+
+where we identify areas to include in the training set by inputing lists as `ann_files` and `scene_idxs`. The train-val split can be simply modified via changing the `train_area` and `test_area` variables.
```
Comment (Tai-Wang) [inline docs/datasets/s3dis_sem_seg.md:301]: where we specify the areas used for training/validation/testing by setting `ann_files` and `scene_idxs` with lists that include corresponding paths.

File: docs/datasets/s3dis_sem_seg.md [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+```
+
+The above exported point cloud file, semantic label file and instance label file are further saved in `.bin` format. Meanwhile `.pkl` info files are also generated for each area. The core function `process_single_scene` of getting data infos is as follows.
+
+```python
+def process_single_scene(sample_idx):
+
+    # save point cloud, instance label and semantic label in .bin file respectively, get info['pts_path'], info['pts_instance_mask_path'] and info['pts_semantic_mask_path']
+
+    print(f'{self.split} sample_idx: {sample_idx}')
```
Comment (ZwwWayne) [inline docs/datasets/s3dis_sem_seg.md:110]: This part of the code seems not to contain tricky things, which can be committed in this documentation.

Comment (Wuziyi616) [reply docs/datasets/s3dis_sem_seg.md]: Yes, I have remove this code block.

**Review by Tai-Wang — APPROVED** [review]

**Pull request merged**